### DISCUSSION TITLE
Sometimes tests fail with timeout

### DISCUSSION THREAD

**Discussion by RedX2501** [webdriverio-community/wdio-electron-service#736] [General]
I cannot pinpoint exactly what's the cause and I'm looking more for pointers on how to debug this instead of a solution. I'm not even sure if this is a wdio or a electron-service problem.

Some tests fail with the message:

```
[0-0] Error in "Electron Testing.File specific tests.testNamel"
Error: Timeout
    at listOnTimeout (node:internal/timers:573:17)
    at processTimers (node:internal/timers:514:7)
[0-0] 2024-08-21T08:17:45.416Z INFO HtmlReporter: onTestFail: 0-0:test-11-39
```

The same test passes other times.

The code for this test is executed many times with different inputs (parameterized) and only fails in a non-reproducible way sometimes.

Without any more information.

WebdriveIO: 8.40.3
wdio-electron-service: 6.6.1
electron: 29.1.4

**Comment by goosewobbler** [comment]
Have you tried enabling [debug mode](https://github.com/webdriverio-community/wdio-electron-service/blob/main/docs/common-issues-debugging.md#common-issues--debugging) for the service?  It's difficult to tell what's going on here with just a timeout error and some version numbers.

There's a lot of fixes in WDIO v9, including a memory leak fix for `execute` / `executeAsync` which the service uses heavily.  I'd recommend focussing on the upgrade in the first instance, let's convert this to a discussion in the meantime.

**Reply by RedX2501** [reply]
I have tried but I kept developing the tests yesterday and now I'm not seeing the timeout anymore.

I'll get back to this thread once it starts appearing again.

Thanks for the suggestions and quick help though.

**Reply by RedX2501** [reply]
I'm receiving them again.

The command described in the help does not work on windows, so I set the environment variable `DEBUG=wdio-electron-service`
and ran `npm run wdio`.

Additionally I set `outputDir` in `wdio.conf.ts`. I also turned `logLevel` to `debug`.

Interestingly this message does not get written to the log files, it still appears in the console. What does this mean? Who is then generating this message?

I see no new output with DEBUG set. Maybe setting it like that is not enough. Is there an alternative? Should I use `process.env`?

The log file `wdio-0-0.log` is huge (hundreds of megabytes) yet contains nothing about timeout errors.

Any idea what I can try next?

**Reply by goosewobbler** [reply]
> I'm receiving them again.
> 
> The command described in the help does not work on windows, so I set the environment variable `DEBUG=wdio-electron-service` and ran `npm run wdio`.
> 
> Additionally I set `outputDir` in `wdio.conf.ts`. I also turned `logLevel` to `debug`.
> 
> Interestingly this message does not get written to the log files, it still appears in the console. What does this mean? Who is then generating this message?

What message are you talking about?  The timeout message?  It would make sense if that message is not in the WDIO logs because it's not a WDIO error, rather an internal node one.  

> I see no new output with DEBUG set. Maybe setting it like that is not enough. Is there an alternative? Should I use `process.env`?
> 
> The log file `wdio-0-0.log` is huge (hundreds of megabytes) yet contains nothing about timeout errors.

The reason for looking at the logs is to expose what processes or tasks WDIO might be getting stuck on or repeating.  If the logs are that huge then it sounds like WDIO is stuck in a loop trying to do something over and over.
 
> Any idea what I can try next?

Apart from looking for loops in the log you could try it on a different OS if you have access, to see if it's a Windows specific issue.  Also check the main WDIO discussion forum as it's a lot more active than this one.
